### PR TITLE
Continue PR #735 (Add support for more constant expressions)

### PR DESCRIPTION
**Pull request by gmarkall**
This continues PR #735, which I think was good to merge, save for the following required additions included in this PR:

- Merge master
- Resolve conflicts
- Document the new methods.

Documentation for the new methods follows the documentation of similar methods in the IR builder documentation (i.e. https://llvmlite.readthedocs.io/en/latest/user-guide/ir/ir-builder.html) - it is not new as such, but instead the wording was re-used for consistency.

### CODE REVIEW

**Comment by gmarkall** [comment]
@esc / @stuartarchibald If one of you is happy with my commit 010232f, then I think this should be good to move to RTM status (based on my previous review of #735).

File: docs/source/user-guide/ir/values.rst [hunk 1]
```diff
@@ -23,6 +23,175 @@ A :ref:`module` consists mostly of values.
...
+   * .. method:: bitcast(typ)
+
+      Convert this pointer constant to a constant of the given pointer type
+      `typ`.
+
+   Floating-point casts:
+
+   * .. method:: fptrunc(typ)
+
+      Truncate (approximate ) floating-point value `self` to floating-point
```
Comment (stuartarchibald) [inline docs/source/user-guide/ir/values.rst:158]:
```suggestion
      Truncate (approximate) floating-point value `self` to floating-point
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for continuing this PR @gmarkall. Given your comments/review already on https://github.com/numba/llvmlite/pull/795#issue-1062487183 I've concentrated on the doc patches, but have also looked at the code changes. The docs seem fine (except a typo) as do the proposed code change. Thanks for your work on this @john-tornblom and @gmarkall.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes!

**Comment by esc** [comment]
@stuartarchibald @gmarkall @john-tornblom thank you for the patches and reviews!

**Comment by john-tornblom** [comment]
Sorry for the late reply, I have been busy with other things the last couple of days. 
Thanks for the rebase @gmarkall, and for a pleasant community engagement experience in general.

**Comment by esc** [comment]
> Sorry for the late reply, I have been busy with other things the last couple of days. Thanks for the rebase @gmarkall, and for a pleasant community engagement experience in general.

No worrries! And thank you again for your contribution, it is much appreciated!

**Pull request merged**